internal/store/group.go

```diff
@@ -81,6 +81,7 @@ func (sqlStore *SQLStore) GetUnlockedGroupsPendingWork() ([]*model.Group, error)
 			Select("ID").
 			From("Installation").
 			Where("GroupID = ?", rawGroup.ID).
+			Where("State = ?", model.InstallationStateStable).
```

Comment:
Will this preclude marking an Installation as needing work when it is in the e.g. `provisioning-failed` state?

Reply:
Yeah, this means that installations in other states won't mark the group as needing work. The supervisors should continue to push those installations to stable eventually meaning it will then get picked up by this check at that time.